OpenRA.Mods.RA/Traits/Disguise.cs

```diff
@@ -48,10 +48,16 @@ public Player Owner
 		{
 			get
 			{
-				if (disguise.Disguised)
-					return self.Owner == self.World.LocalPlayer ? self.Owner : disguise.AsPlayer;
+				if (!disguise.Disguised)
+					return self.Owner;
 
-				return self.Owner;
+				Player p = null;
+				if (self.World.LocalPlayer != null && self.World.LocalPlayer.WinState == WinState.Undefined)
```

Comment:
When I thought about this the other day, I figured that there's just two fundamental cases
* when `RenderPlayer.InternalName == "Everyone" || self.Owner.IsAlliedWith(RenderPlayer)` then you ought to see through the disguise (allied case: allies in game, allies spectators, everyone spectator, players with disable shroud cheat),
* and otherwise just show the disguise (enemy case).